src/test/java/net/lingala/zip4j/crypto/PBKDF2/BinToolsTest.java

```diff
@@ -0,0 +1,65 @@
+package net.lingala.zip4j.crypto.PBKDF2;
+
+import java.lang.reflect.Array;
+import net.lingala.zip4j.crypto.PBKDF2.BinTools;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.Timeout;
+
+public class BinToolsTest {
+  @Rule
+  public final Timeout globalTimeout = new Timeout(10000);
+
+  @Rule
+  public final ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void bin2hexInput1OutputNotNull2() {
+    final byte[] b = {(byte) 112};
+    Assert.assertEquals("70", BinTools.bin2hex(b));
+  }
+
+  @Test
+  public void bin2hexInput0OutputNotNull() {
+    final byte[] b = {};
```

Comment:
This variable is not necessary. You can directly do `Assert.assertEquals("", BinTools.bin2hex(new byte[] {}));` as you did in other methods below.